test/CoreTypechecker.hs

```diff
@@ -263,48 +295,63 @@ kAppICompTy =
 -- ((Nat -> Nat -> Nat) -> (Nat -> Nat -> Nat) -> (Nat -> Nat -> Nat)) ->
 -- ((Nat -> Nat -> Nat) -> Nat -> Nat -> Nat)
 -- (Nat -> Nat -> Nat)
-scombinator ∷ ∀ primTy primVal. IR.Term primTy primVal -- S = \x.\y.\z. (xz) (yz)
-scombinator =
-  IR.Lam --x (Bound 2)
-    ( IR.Lam --y (Bound 1)
-        ( IR.Lam --z (Bound 0)
-            ( IR.App
-                (IR.Ann
-                    (SNat 1)
-                    ( IR.App
-                        (IR.Ann
-                            (SNat 1)
-                            (IR.Bound 2)
-                            () -- Annotation of x
+
+-- SKI1 = K 1 (I 1)
+-- the first input K takes inputs 1 and 1 so has type signature of Nat -> Nat -> Nat
+-- the second input I has type signature Nat -> Nat
+-- the third input is Nat
+-- type signature of this S is (Nat -> Nat -> Nat) -> (Nat -> Nat) -> Nat -> Nat
+ski1 ∷ NatTerm -- ∀ primTy primVal. IR.Term primTy primVal -- S = \x.\y.\z. (xz) (yz)
+ski1 =
```

Comment:
It seems to me that `ski1` corresponds to just the `S`, rather than the whole `S K I 1` expression? It looks like the AST for

```
λxyz. (1 ((1 x : 1 ℕ → 0 ℕ → ℕ) z) : 0 ℕ → ℕ)
      ((1 y : 1 ℕ → ℕ) z)
```

i.e. `S`, but `test_ski1` below is asserting that this has the type `ℕ` (the type of the whole `S K I 1`).

Reply:
Thank you! You are right! I've been trying to annotate `s` but I think I got the levels wrong somehow. See [here](https://gist.github.com/thealmarty/f652a74ead0c84c4b8c58b7bcf54a8c6). Would like to borrow your and @cwgoes 's eyes - my eyes are kinda done - sorry I don't know how to make this easier to read. Recommendations welcome!